util/perl/OpenSSL/config.pm

```diff
@@ -322,6 +322,7 @@ sub determine_compiler_settings {
 
             # If we got a version number, process it
             if ($v) {
+                $v =~ s/[^.]*.[0B]*// if $SYSTEM eq 'HP-UX';
```

Comment:
I am not 100% sure about this one, used from the above hunk, one needs to clean up: `aCC: HP C/aC++ B3910B A.06.29 [Oct 18 2016]`. I think that `s/[^.]*.[0]*//;` would be enough.